connector_amazon/models/amazon_sale_importer.py

```diff
@@ -3,14 +3,204 @@
+            'price_unit': float(line['item-price']) + float(line['item-tax']),
+            'shipping': float(line['shipping-price']) + \
+            float(line['shipping-tax']),
+        }
+
+    def _create_sales(self, sales, meta_attachment):
+        """ We process all sale orders of the file
+        """
+        backend = meta_attachment.amazon_backend_id
+        for order_item, sale in sales.items():
+            partner, part_ship = self._get_partners(sale)
+            vals = {
+                'name': '%s %s' % (
+                    backend.sale_prefix or '',
+                    sale['sale']['origin']),
+                'partner_id': partner.id,
+                'partner_shipping_id': part_ship.id,
+                'pricelist_id': backend.pricelist_id.id,
+                'external_origin': 'ir.attachment.metadata,%s'
+                % meta_attachment.id,
+                'origin': sale['sale']['origin'],
+            }
+            if sale.get('lines'):
+                ship_price = self._complete_products(sale['lines'], backend)
+                vals['order_line'] = [
+                    (0, '_', {key: val for key, val in line.items()})
+                    for line in sale['lines']]
+                if ship_price:
+                    # import pdb; pdb.set_trace()
+                    ship_vals = {
+                        'product_uom_qty': 1,
+                        'price_unit': ship_price,
+                        'product_id': backend.shipping_product.id,
+                    }
+                    vals['order_line'].append((0, '_', ship_vals), )
+            self.env['sale.order'].create(vals)
+
+    def _get_partners(self, sale):
+        """ We have to search or create partners: main and shipping partner
+        """
+        partner_m = self.env['res.partner']
+        partner = partner_m.search(
+            [('email', '=', sale['partner']['email'])])
+        part_ship = False
+        sale['part_ship']['country_id'], sale['part_ship']['state_id'] = \
+            self._get_state_country(sale)
+        if partner:
+            if sale['part_ship'].get('street3') and 'street3' not in \
+                    partner_m._fields:
+                # if street3 doesn't exist in odoo: according to modules
+                sale['part_ship']['street2'] = '%s %s' % (
+                    sale['part_ship']['street2'], sale['part_ship']['street3'])
+            part_ship = partner_m.search([
+                (fieldname, '=', val)
+                for fieldname, val in sale['part_ship'].items()
+                if fieldname in partner_m._fields])
+        else:
+            partner = partner_m.create(sale['partner'])
+        if not part_ship:
+            sale['part_ship']['parent_id'] = partner.id
+            vals = {k: v for k, v in sale['part_ship'].items()
+                    if k in partner_m._fields}
+            part_ship = partner_m.create(vals)
+        return (partner[0], part_ship[0])
```

Comment:
We plan to add a dependency to https://github.com/OCA/partner-contact/issues/404

@paulius-sladkevicius could you give your opinion.